sdk/storage/azure-storage-perf/src/main/java/com/azure/storage/blob/perf/core/BlobTestBase.java

```diff
@@ -50,6 +64,39 @@ public BlobTestBase(TOptions options) {
 
         blockBlobClient = blobClient.getBlockBlobClient();
         blockBlobAsyncClient = blobAsyncClient.getBlockBlobAsyncClient();
+
+        this.createBlob = createBlob;
+        this.singletonBlob = singletonBlob;
```

Comment:
for the sake of simplicity I'd just remove `singletonBlob` and always create a blob per test.
For downloads it doesn't matter much and it' just making the code more complicated.

Reply:
@mikeharder specifically called out that download tests should only upload one blob for the whole run and all parallel instances should target downloading the same blob. Mike, do you have input here?

Reply:
I don't feel too strongly either way.  Uploading a single blob makes setup a little faster (especially for `1GB parallel=8`), but it also makes the code a little more complex.  I do think it's nice if this is consistent across languages though, and the other languages currently upload a single blob.

Reply:
btw. creating abstract base for upload and downloads should give flexibility to preserve the behavior without much complexity.